plugins/esi/combo_handler.cc

```diff
@@ -1052,21 +1095,86 @@ getContentType(TSMBuffer bufp, TSMLoc hdr_loc, string &resp_header_fields)
     int n_values = TSMimeHdrFieldValuesCount(bufp, hdr_loc, field_loc);
     for (int i = 0; i < n_values; ++i) {
       value = TSMimeHdrFieldValueStringGet(bufp, hdr_loc, field_loc, i, &value_len);
-      if (!values_added) {
-        resp_header_fields.append("Content-Type: ");
-        values_added = true;
-      } else {
-        resp_header_fields.append(", ");
+      ts::TextView tv{value, value_len};
+      tv              = tv.prefix(';').rtrim(std::string_view(" \t"));
+      auto const &ctw = _content_type_whitelist;
+      if (ctw.empty()) {
+        ;
+      } else if (std::find_if(ctw.begin(), ctw.end(), [tv](ts::TextView tv2) -> bool { return strcasecmp(tv, tv2) == 0; }) ==
+                 ctw.end()) {
+        return false;
+      } else if (tv.empty()) {
+        // Whitelist is bad, contains an empty string.
+        return false;
+      }
+      if (!_added_content_type) {
+        if (!values_added) {
+          _resp_header_fields.append("Content-Type: ");
+          values_added = true;
+        } else {
+          _resp_header_fields.append(", ");
+        }
+        _resp_header_fields.append(value, value_len);
       }
-      resp_header_fields.append(value, value_len);
     }
     TSHandleMLocRelease(bufp, hdr_loc, field_loc);
     if (values_added) {
-      resp_header_fields.append("\r\n");
-      retval = true;
+      _resp_header_fields.append("\r\n");
+
+      // Assume that the Content-type field from the first header covers all the responses being combined.
+      _added_content_type = true;
     }
+    return true;
   }
-  return retval;
+  // No content type header field so doesn't pass whitelist if there is one.
+  return _content_type_whitelist.empty();
+}
+
+void
+ContentTypeHandler::loadWhiteList(std::string const &file_spec)
+{
+  std::fstream fs;
+  char line_buffer[256];
+  bool extra_junk_on_line{false};
+  int line_num = 0;
+
+  fs.open(file_spec);
+  if (fs.good()) {
+    for (;;) {
+      ++line_num;
+      fs.getline(line_buffer, sizeof(line_buffer));
+      if (!fs.good()) {
+        break;
+      }
+      std::string_view bs{" \t"sv};
```

Comment:
You use this several places, you should make it a `constexpr` value to ensure consistency.